packages/ui-date-input/src/DateInput2/index.tsx

```diff
@@ -214,21 +214,21 @@ const DateInput2 = ({
     return date ? [formatDate(date), date.toISOString()] : ['', '']
   }
 
-  const formatDate = (date: Date): string => {
+  const formatDate = (date: Date, timeZone: string = getTimezone()): string => {
     // use formatter function if provided
     if (typeof dateFormat !== 'string' && dateFormat?.formatter) {
       return dateFormat.formatter(date)
     }
     // if dateFormat set to a locale, use that, otherwise default to the user's locale
     return date.toLocaleDateString(
       typeof dateFormat === 'string' ? dateFormat : getLocale(),
-      { timeZone: getTimezone(), calendar: 'gregory', numberingSystem: 'latn' }
+      { timeZone, calendar: 'gregory', numberingSystem: 'latn' }
     )
   }
 
   const getDateFromatHint = () => {
     const exampleDate = new Date('2024-09-01')
-    const formattedDate = formatDate(exampleDate)
+    const formattedDate = formatDate(exampleDate, 'UTC') // exampleDate is in UTC so format it as such
```

Comment:
In the date constructor above it would be nicer to explicitly use the UTC timezone, not to rely on a JS quirk, e.g. `Date.UTC('2024-09-01')`